PYTHON/activityLog/activityLog.py

```diff
@@ -0,0 +1,356 @@
+		# if run by user.
+		# Explained: sys.stdin will be a TTY if run by console (by the user).
+		if os.isatty(sys.stdin.fileno()):
+			self.cronjob = False
+			self.silent_mode = False
+		else:
+			self.cronjob = True
+			self.silent_mode = True
+
+
+	# Returns a tuple of the stdout and stderr of a command. 
+	# NOTE that the use of shell=True is extremely dangerous from a security standpoint. Using it
+	# here because the alternative (parsing each command and argument into a list) makes piping
+	# multiple commands difficult.
+	def run_command(self, cmd):
+		return subprocess.Popen(
+			cmd, 
+			shell=True, 
+			stdout = subprocess.PIPE, # stdout in [0]
+			stderr = subprocess.PIPE).communicate() # stderr in [1]
+
+	def dateString(self, epoch = False):
+		# returns time from epoch.  ON UNIX, it is 0 hours on Jan. 1st, 1970
+		if epoch == True: 
+			return time.time()
+		elif epoch != True:
+		# using UTC (Similar to GMC)
+		# strftime converts the struct_time format into a string. see documentation
+			dateTime = datetime.utcnow().strftime("%mm_%dd_%Yy__at_UTC_%Hh_%Mm_%Ss")
+			return dateTime
+
+
+	def rootFsRwRoStateCheck(self):
+	# Checks to see if root (/) directory is read/write 
+		try:
+			# Same output as running "mount" in BASH 
+			with open("/proc/mounts", "r") as f:
+				for line in f:
+					# Look for line with root (/) dir and "rw" in the line
+					if " / " in line and "rw" in line:
+						return True
+				return False
+		except:
+			if self.silent_mode == "False":
+				print("Opening /proc/mounts failed.  Returning False")
+				return False
+			else:
+				return False
+
+
+	def topRC(self):
+		# Custom ~./toprc config.  I have no idea what is happening here.
+
+		# Expands "~" to user home dir
+		toprc_loc = os.path.expanduser("~/.toprc")
+		toprcexists = os.path.isfile(toprc_loc)
+		toprc = """RCfile for \"top with windows\"
+	Id:a, Mode_altscr=0, Mode_irixps=1, Delay_time=3.000, Curwin=0
+	Def     fieldscur=AEHIOQTWKNMbcdfgjplrsuvyzX
+	        winflags=30137, sortindx=10, maxtasks=0
+	        summclr=1, msgsclr=1, headclr=3, taskclr=1
+	Job     fieldscur=ABcefgjlrstuvyzMKNHIWOPQDX
+	        winflags=62777, sortindx=0, maxtasks=0
+	        summclr=6, msgsclr=6, headclr=7, taskclr=6
+	Mem     fieldscur=ANOPQRSTUVbcdefgjlmyzWHIKX
+	        winflags=62777, sortindx=13, maxtasks=0
+	        summclr=5, msgsclr=5, headclr=4, taskclr=5
+	Usr     fieldscur=ABDECGfhijlopqrstuvyzMKNWX
+	        winflags=62777, sortindx=4, maxtasks=0
+	        summclr=3, msgsclr=3, headclr=2, taskclr=3"""
+
+		if toprcexists:
+			shutil.move(toprc_loc, os.path.expanduser("~/.backup_toprc"))
+		try:
+			with open(toprc_loc, 'w') as f:
+				f.write(toprc)
+				f.close()
+		except:
+			if self.silent_mode == False:
+				print("Cannot write to ~/.toprc.")
+
+	def getServerLoad_BASH(self):
+	# OLD BASH WAY OF GETTING SERVER LOAD 
+		cmd = ("""uptime | grep -Pio "average\:(\s\d{1,}\.\d{1,}\,){1,}(\s\d{1,}\.\d{1,})" |"""
+		"""sed -r "s~(average\:\s)~~g" |"""
```

Comment:
I would use plain `'strings'` here instead of `"""strings"""` if you are going to use the Python magic multiline string catenation trick.  I am guessing that this started out as a big multiline `"""string"""` and then evolved when it turned out that the newlines were breaking things in the shell?